commonknowledge/meep-intelligence-hub
# MEEP-132 Install auth on Mapped NextJS app

## Description
This PR 
- adds a fresh installation of NextJs v13 into the repo 
- adds launch settings to run frontend and backend with play button in run and debug
- sets up a basic authentication system with JWT

## Motivation and Context
Addresses issue [MEEP-132](https://linear.app/commonknowledge/issue/MEEP-132/install-auth-on-mapped-nextjs-app)

## How Can It Be Tested?
Download the branch and run by pressing play on App in the VSCode Run and Debug panel
Open up a shell inside the Django web container by running:
```
cd django
docker compose exec web bash
```
then run 
`python manage.py createsuperuser `
to create an account

You should be able to login then at http://localhost:8000/accounts/login/

Go to the Django admin and make sure your user status is verified
http://localhost:8000/admin/gqlauth/userstatus/

![Untitled](https://github.com/commonknowledge/meep-intelligence-hub/assets/43313455/9a565491-b6e6-48f4-92d9-1858bb618ead)

Create an .env.local file in the root of the project and add the graphql backend url
`NEXT_PUBLIC_BACKEND_BASE_URL="http://localhost:8000/graphql"`

Go to http://localhost:3000/login fill out the form to login. Use the same credentials as with Django

You should be redirected to http://localhost:3000/account

Click on the logout page on the menu and click the button to logout

## Types of changes
